mailserver/mailserver.go

```diff
@@ -164,6 +164,16 @@ func (s *WMailServer) Archive(env *whisper.Envelope) {
 		log.Error(fmt.Sprintf("rlp.EncodeToBytes failed: %s", err))
 		archivedErrorsCounter.Inc(1)
 	} else {
+		// Recover from possible goleveldb panics
+		defer func() {
+			if r := recover(); r != nil {
+				if errString, ok := r.(string); ok {
+					err := fmt.Errorf("recovered from panic in Archive: %v", errString)
+					log.Error(err.Error())
```

Comment:
`log.Error()` takes `string` as an argument, so no need to create error interface here.